openedx/features/course_experience/static/course_experience/js/currency.js

```diff
@@ -0,0 +1,83 @@
+import whichCountry from 'which-country';
+import 'jquery.cookie';
+import $ from 'jquery'; // eslint-disable-line import/extensions
+
+export class Currency {  // eslint-disable-line import/prefer-default-export
+
+  setCookie(countryCode, l10nData) {
+    function pick(curr, arr) {
+      const obj = {};
+      arr.forEach((key) => {
+        obj[key] = curr[key];
+      });
+      return obj;
+    }
+    const userCountryData = pick(l10nData, [countryCode]);
+    let countryL10nData = userCountryData[countryCode];
+
+    if (countryL10nData) {
+      countryL10nData.countryCode = countryCode;
+    } else {
+      countryL10nData = {
+        countryCode: 'USA',
+        symbol: '$',
+        rate: '1',
+        code: 'USD',
+      };
+    }
+    this.countryL10nData = countryL10nData;
+    $.cookie('edx-price-l10n', JSON.stringify(countryL10nData), {
+      expires: 1,
+    });
+  }
+
+  setPrice() {
+    const l10nCookie = this.countryL10nData;
+    const lmsregex = /(\$)(\d*)( USD)/g;
+    const price = $('input[name="verified_mode"]').filter(':visible')[0];
+    const regexMatch = lmsregex.exec(price.value);
+    const dollars = parseFloat(regexMatch[2]);
+    const converted = dollars * l10nCookie.rate;
+    const string = `${l10nCookie.symbol}${Math.round(converted)} ${l10nCookie.code}`;
+    // Use regex to change displayed price on track selection
+    // based on edx-price-l10n cookie currency_data
+    price.value = price.value.replace(regexMatch[0], string);
+  }
+
+  getL10nData(countryCode) {
+    const l10nData = JSON.parse($('#currency_data').attr('value'));
+    if (l10nData) {
+      this.setCookie(countryCode, l10nData);
+    }
+    this.setPrice();
+  }
+
+  getCountry(position) {
```

Comment:
If you remove the `this.setPrice()` call in `getL10nData` I think you can simplify `getCountry`

```javascript
getCountry(position) {
    const countryCode = whichCountry([position.coords.longitude, position.coords.latitude]);
    this.countryL10nData = JSON.parse($.cookie('edx-price-l10n'));

    if (countryCode 
        && !(this.countryL10nData 
                && this.countryL10nData.countryCode === countryCode)
      ) {
        this.getL10nData(countryCode);
      }
    }
    this.setPrice();
  }
```

Reply:
changed